### ISSUE TITLE
Center atom for improper dihedral coefficients in oplsaa.it

### ISSUE THREAD

**Issue by HUANGZR86** [jewettaij/moltemplate#79]
Dear developer, 

I am using the OPLS-AA parameters from the file oplsaa.lt. May I ask for the improper dihedral X_X_003_004, which one should be the center atom?

![image](https://user-images.githubusercontent.com/44540970/178705683-ed871e57-8b2b-462a-9cb5-d36b05720a09.png)

Best regards,
Huang Ziru

**Comment by jewettaij** [comment]
Hi Huang

I think that for OPLSAA in moltemplate, the 3rd atom is the center atom.  *(Note: This atom is named "2" in the diagram below, because the numbers begin at 0 not 1.)*
*(To be clear, this applies to the "Impropers" section of the "system.data" file generated by moltemplate.)*

To create the "oplsaa.lt" file, I converted the "oplsaa.prm" file which comes with TINKER.  I think that TINKER also uses the 3rd atom.

Years ago, several moltemplate users and I spent days trying to confirm that this is the correct order.  But this was impossible to confirm because some molecule builders (like BOSS and Materials Studio) do not document how the atom order is determined or where the central atom is.  For this reason, we could not always reproduce the atom order created by these programs.

If this sounds like a serious problem.  However perhaps it is not a problem:  Fortunately, all of the improper interactions in OPLSAA are even functions((U(φ)=U(-φ)), and U(φ) has a minima at φ=0, which only occurs when the 4 atoms are coplanar (in the same plane).  The important thing is that all 4 atoms should nearly belong to the same plane.  It turns out that the atom order (and the spring constant used to enforce coplanarity) is not very important because the flexibility of most molecules is a consequence of the dihedral interactions and 3-body "angle" interactions.  So even if the improper interactions "oplsaa.lt" are too strong, all 4 atoms will still be nearly coplanar, and this should not affect the flexibility of the molecule very much.

Perhaps I am wrong?

If you have discovered we made a mistake in the OPLSAA, then please let me know.

If you are curious how moltemplate works, see the ["Impropers by Type" section of the "oplsaa.lt" file.](https://github.com/jewettaij/moltemplate/blob/c76cb98a817c1c7fc5b0f8b926b72fa032709dcd/moltemplate/force_fields/oplsaa.lt#L8207)
Notice the "(opls_imp.py)".  This tells moltemplate that the ["opls_imp.py" file](https://github.com/jewettaij/moltemplate/blob/c76cb98a817c1c7fc5b0f8b926b72fa032709dcd/moltemplate/nbody_alt_symmetry/opls_imp.py#L7) contains the rules for generating improper interactions.  Here is an excerpt of the contents of the that file:
```
#    To find 4-body "improper" interactions,
#    (by default, most of the time), we would use this subgraph:
#           0
#           *                  1st bond connects atoms 2 and 0
#           |              =>  2nd bond connects atoms 2 and 1
#         _.*._                3rd bond connects atoms 2 and 3
#       *'  2  `*
#      1         3
#
# In OPLS (as implemented in TINKER .PRM files), the central atom
# is the third atom ("2").
bond_pattern = Ugraph([(2,0), (2,1), (2,3)])
```

**Comment by jewettaij** [comment]
Since the 2023 version of OPLSAA was released (v2.21.2), the answer to this question is no longer true.
For this new 2023 version of oplsaa.lt, the first atom is the central atom.  The symmetry rules have changed too.

So this question is no longer relevant, and my answer is misleading.  I will close this issue.
